### DISCUSSION TITLE
Should we enable negative margins by default?

### DISCUSSION THREAD

**Discussion by MartijnCuppens** [twbs/bootstrap#30649] [General]
We introduced negative margins in 4.2 to be able to use responsive gutters. Since we have dedicated gutter classes in our new grid system, there wasn't a need for these classes anymore, at least not for the gutter widths. Therefor I disabled them by default in https://github.com/twbs/bootstrap/pull/30585.

What are your thoughts on this, should we bring these classes back or keep the current situation where you can enable them with the `$enable-negative-margins: true` setting?

**Comment by MartijnCuppens** [comment]
Removing the classes decreased the file size about 1kb gzipped (from 21839 bytes to 20858 bytes, tested April 14th). We also generate `.mx-n*` classes, classes that have little use cases imo.

Because the drop in file size is pretty weighty, some (if not all) classes do not have lots of use cases, and because it's quite easy to enable the classes with `$enable-negative-margins: true`, I think it's better to leave it as is.

**Reply by ffoodd** [reply]
I agree—moreover if one need a specific negative margin utility (instead of all), it can be easily extended through the utility API, can't it?

**Reply by MartijnCuppens** [reply]
yup indeed

**Comment by XhmikosR** [comment]
The point isn't what can be enabled or not via the API. The point is what we offer out of the box. But maybe only a few people use negative margins, which TBH is a bad practice most of the time.

BTW I'm not a fan of the discussions since I don't get notified.
